Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,297 @@
+    LogMsg "Checking for preconditions"
+
+    if [ $EUID -ne 0 ]; then
+        LogErr "This script must be run as root: $EUID"
+        SetTestStateAborted
+        exit 0
+    fi
+
+    verify_distro
+
+    if check_da_directories_exist; then
+        SetTestStateAborted
+        exit 0
+    fi
+
+    LogMsg "Preconditions met"
+}
+
+function download_da_linux_installer() {
+    LogMsg "Download Dependency Agent Linux installer"
+
+    if ! curl -L https://aka.ms/dependencyagentlinux -o "$da_installer"; then
+        fail_test Download failed
+    fi
+
+    chmod +x "$da_installer"
+}
+
+function verify_install_da() {
+    LogMsg "Starting Install tests"
+
+    "$da_installer" -vme
+
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        fail_test "Install failed with exit code ${ret}"
+    fi
+    LogMsg "Dependency Agent installed successfully"
+
+    verify_expected_file "$uninstaller"
+    verify_expected_file "/etc/init.d/microsoft-dependency-agent"
+    verify_expected_file "$da_log_dir/install.log"
+
+    bin_version=$("$da_installer" --version | awk "{print $5}")
+    install_log_version=$(sed -n "s/^Dependency Agent version\.revision: //p" $da_log_dir/install.log)
+    if [ "$bin_version" -ne "$install_log_version" ]; then
+        fail_test "Version mismatch between bin version($bin_version) and install log version($install_log_version)"
+    fi
+    LogMsg "Version matches between bin version and install log version"
+
+    LogMsg "Install tests completed"
+}
+
+function verify_uninstall_da() {
+    LogMsg "Starting uninstall tests"
+
+    $uninstaller
+
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        fail_test "Uninstall failed with exit code ${ret}"
+    else
+        LogMsg "Uninstalled with exit code 0"
+    fi
+
+    if check_da_directories_exist; then
+        fail_test "Directory exists"
+    fi
+
+    LogMsg "Uninstall tests completed"
+}
+
+function generate_network_activity() {
+    LogMsg "Generate network activity"
+    watch -n 5 curl https://microsoft.com &>/dev/null &
+    network_activity_generator_pid=$!
+}
+
+function enable_disable_da(){
+    LogMsg "Starting Enable/Disable DA tests"
+
+    verify_install_da
+
+    # Wait for 30s for DA to start running and check for service.log
+    sleep 30
+    verify_expected_file "$da_log_dir/service.log"
+
+    # Check for service.log.1 file and confirm service.log.2 doesn't exist
+    service_log_time=$(date -r $da_log_dir/service.log +%s)
+    time_limit=$(($service_log_time + $enable_timeout))
+
+    while [ ! -f "$da_log_dir/service.log.1" ]
+    do
+        current_time=$(date +%s)
+        if [ $current_time -gt $time_limit]; then
+            fail_test "Exceeded time limit to check service.log.1 file"
+        fi
+        sleep 5
+    done
+
+    verify_expected_file "$da_log_dir/service.log.1"
+    
+    if [ -f "$da_log_dir/service.log.2" ]; then
+        fail_test "$da_log_dir/service.log.2 exist."
+    fi
+    
+    if [ ! -c "/dev/msda" ]; then 
```

Comment:
Double checked the test plan. The `/dev/msda` check is supposed to come after the device setup log message check. If the device setup failed, we will want to get the failure code from the log. If we check for `/dev/msda` first the test case will terminate and the details from the log won't be included.

Reply:
Done